src/modules/aws_iam@0.0.1/index.ts

```diff
@@ -0,0 +1,72 @@
+import { Role as AWSRole } from '@aws-sdk/client-iam'
+
+import { Role, } from './entity'
+import { AWS,  } from '../../services/gateways/aws'
+import { Context, Crud, Mapper, Module, } from '../interfaces'
+import * as metadata from './module.json'
+
+export const AwsIamModule: Module = new Module({
+  ...metadata,
+  utils: {
+    roleMapper: async (role: AWSRole, ctx: Context) => {
+      const out = new Role();
+      out.arn = role.Arn;
+      out.name = role.RoleName ?? '';
+      out.policyDocument = role.AssumeRolePolicyDocument ?? '';
+      const client = await ctx.getAwsClient() as AWS;
+      out.policyArns = await client.getRolePolicyArns(out.name);
+      return out;
+    },
+  },
+  mappers: {
+    role: new Mapper<Role>({
+      entity: Role,
+      equals: (a: Role, b: Role) => Object.is(a.name, b.name) &&
+        Object.is(a.arn, b.arn) &&
+        Object.is(a.policyDocument, b.policyDocument) &&
```

Comment:
is policy document a json as string? that could have some trouble if the json does not come always ordered from aws

Reply:
They probably store it as a string on their side (I haven't seen these docs mutated before, at least) but that is a fair point.

I don't think it is a likely problem, though, so maybe a comment that this depends on AWS not regenerating the JSON out from under us?

Reply:
that is a good point. I will add a comment for now and test to see if it might be problematic